tecken/tests/test_sentry.py

```diff
@@ -226,3 +232,38 @@ def test_count_sentry_scrub_error(metricsmock):
     metricsmock.assert_incr(
         "tecken.sentry_scrub_error", value=1, tags=["host:testnode"]
     )
+
+
+def test_sentry_wrap_non_app_error_has_release():
+    port = os.environ.get("EXPOSE_SENTRY_PORT", 8090)
+
+    # Flush fakesentry to ensure we fetch only the desired error downstream
+    requests.post(f"http://fakesentry:{port}/api/flush/")
+
+    current_dir = os.path.dirname(os.path.abspath(__file__))
+    base_dir = os.path.join(os.path.dirname(current_dir), "..")
+    release = get_release_name(base_dir)
+
+    expected_event = {"release": release}
+
+    # Pass a non-Django command that will error to sentry_wrap
+    non_app_command = "ls -2"
+    sentry_wrap_command = f"python bin/sentry_wrap.py wrap-process -- {non_app_command}"
+    cmd_args = shlex.split(sentry_wrap_command)
+    subprocess.run(cmd_args, timeout=10)
+
+    # Avoid a race condition where fakesentry hasn't processed the event yet.
+    actual_event = {}
+    is_condition_met = False
+    while not is_condition_met:
+        time.sleep(1)
+        errors_resp = requests.get(f"http://fakesentry:{port}/api/errorlist/")
+        errors_resp.raise_for_status()
+        if len(errors_resp.json()) > 0:
+            error_id = errors_resp.json()["errors"][0]
+            error_resp = requests.get(f"http://fakesentry:{port}/api/error/{error_id}")
+            error_resp.raise_for_status()
+            actual_event = error_resp.json()["payload"]
+            is_condition_met = True
```

Comment:
I know this is generally an antipattern, but I couldn't find a better way to do this. Based on local testing, this `while` loop will typically only run through once (likely since we're making a request to another local docker container).

Reply:
One nice thing about running `sentry_wrap.py` in a subprocess is that when that process is done, we're guaranteed that sentry has sent the event and it's been handled by Kent and stored. There shouldn't be a race condition anymore.

After reworking the test to run `sentry_wrap.py` in a subprocess, did you continue to have problems with Kent not having any errors, yet?

Reply:
Ah good point, because part of the code in `sentry_wrap.py` handles `kent`'s response. Yep I just ran in 20x locally and don't see the race condition with 0 sleeps. Nice!